### PR TITLE
Comparison modals

### PR DESCRIPTION
**Pull request by milofultz**
This creates the basics of the comparison modal. It iterates through the "features" of the current product and whatever product has been selected out of the related products carousel. 

I have yet to find a product that uses a true/false or yes/no within the features, which is what the `check` graphics are for, so I am unsure of how to fire those off until I find one in the wild.

### CODE REVIEW

File: client/__tests__/Comparison.test.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      }
+    }
+  }
+};
+
+const relatedProductInformation = JSON.parse(JSON.stringify(currentProductInformation));
+relatedProductInformation.features.push({
+  feature: 'this feature is only on the related product',
+  value: true
+});
```
Comment (baudot) [inline client/__tests__/Comparison.test.js:139]: In an ideal world, I'd also remove a feature that was in the original, just so regression tests in the future will make sure that doesn't cause crashes. But that's a pretty small nit to pick.

File: client/__tests__/Comparison.test.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  expect(relatedTitle).toBeInTheDocument();
+});
+
+test('Should contain traits', () => {
+  // This should contain 5 traits, as there are four that are shared and one that is unique
+  const comparisonTraits = screen.getAllByTestId('comparison-characteristics__characteristic');
+  expect(comparisonTraits.length).toBe(5);
+});
+
+test('Should render a check mark if feature attribute is boolean', () => {
```
Comment (baudot) [inline client/__tests__/Comparison.test.js:170]: is boolean? Is present, does this mean? 
This is as close as I get to a change request - this test description is a little opaque.

**Review by baudot — APPROVED** [review]
Even after careful review, the only thing that occurs to me to suggest is a tweak to one test. Nothing worth holding this up over.

File: client/__tests__/Comparison.test.js (hunk 1 above)
Comment (milofultz) [reply client/__tests__/Comparison.test.js:139]: This is a good idea, thank you. I'll add this as a ticket

File: client/__tests__/Comparison.test.js (hunk 2 above)
Comment (milofultz) [reply client/__tests__/Comparison.test.js:170]: Good point. They are naturally just a string, but they mention in the doc that if the attribute is true or false/yes or no, use a check mark. I have yet to experience one, so am assuimng this is how they will be presented, but I'll clarify the tests further for future readers.

**Pull request merged**